### DISCUSSION TITLE
UI Flickerng in WSL Ubuntu

### DISCUSSION THREAD

**Discussion by mail2mhossain** [AvaloniaUI/Avalonia#9994] [Q&A] [answered]
I have an Avalonia UI desktop project. In "OnFrameworkInitializationCompleted" of App.axaml.cs class, I have following code

public override void OnFrameworkInitializationCompleted ()
        {
            if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
            {
                MainWindowViewModel viewModel= new MainWindowViewModel();
                MainWindow window = new MainWindow
                {
                    DataContext = viewModel,
                };
                viewModel.MainWindow= window;
                viewModel.ShowUI(false);
                desktop.MainWindow= window;
            }

            base.OnFrameworkInitializationCompleted();
        }

In viewModel.ShowUI(false), i am hiding the window.

It is working fine in Windows, but flickering in WSL Ubuntu. 

[Avalonia_Test.zip](https://github.com/AvaloniaUI/Avalonia/files/10302072/Avalonia_Test.zip)

**Comment by emmauss** [comment]
I tried your test on wsl ubuntu 22.04.0 lts. The window appeared for 1 frame, then disappears. It doesn't appear any longer. Is this what you are experiencing? On windows, the window is completely hidden from launch.

**Comment by mail2mhossain** [comment]
The window appeared for 1 frame, then disappears. Is this what you are experiencing?  - YES. 

I want same behavior as Windows ( the window is completely hidden from launch).

**Comment by mail2mhossain** [comment]
How can I get same behavior in Ubuntu? Should I wait for a fix?

**Comment by kekekeks** [comment]
You are asking the window to show and then to hide. Which is what happens.
If you don't need to window to be shown, don't show it in the first place.

**Comment by mail2mhossain** [comment]
I am not showing the window. The framework shows windows automatically. I am trying not to show but hide.

**Comment by kekekeks** [comment]
Don't set it as MainWindow then.

**Comment by mail2mhossain** [comment]
If we do not set MainWindow, then the app will close when any of window is closed by a user. 

I have attached a test project in my first post. If you run the project, you will get avalonia icon in system tray. Right click on it, you will get a menu. Click on "Setting Window", it will appear. If you close the UI, App will still run.

But if we do not set any MainWindow, then App will close after closing any UI.

**Comment by timunie** [comment, marked as answer]
@mail2mhossain I think you can control this behavior in the ApplicationLifetime: 

https://docs.avaloniaui.net/docs/getting-started/application-lifetimes
https://reference.avaloniaui.net/api/Avalonia.Controls.ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime/A003FA66
https://stackoverflow.com/questions/71256272/how-can-we-change-the-default-shutdown-mode-of-an-avalonia-application

**Comment by mail2mhossain** [comment]
Thank you very much. It is working now.

**Comment by timunie** [comment]
Converted this to Q&A as I think this is not a bug, but a question. Let me know if that was wrong.
